t/ui/21-admin-needles.t

```diff
@@ -111,6 +112,59 @@ like(
     "redirected to right module too"
 );
 
+subtest 'dereference symlink when displaying needles info' => sub {
+    my $real_needle_dir    = getcwd . '/t/data/openqa/share/tests/opensuse';
+    my $symlink_needle_dir = getcwd . '/t/data/openqa/share/tests/test_symlink_dir';
+    symlink($real_needle_dir, $symlink_needle_dir) or die "Cannot make symlink $!";
+    my $needle_dir            = $schema->resultset('NeedleDirs');
+    my $symlink_needle_dir_id = $needle_dir->create(
+        {
+            path => $symlink_needle_dir . '/needles',
+            name => 'symlink_needle_dir'
+        })->id;
+    my $real_needle_dir_id = $needle_dir->create(
+        {
+            path => $real_needle_dir . '/needles',
+            name => 'real_needle_dir'
+        })->id;
+    my $needles        = $schema->resultset('Needles');
+    my $symlink_needle = $needles->create(
+        {
+            dir_id                 => $symlink_needle_dir_id,
+            filename               => 'bootloader.json',
+            last_seen_module_id    => undef,
+            last_seen_time         => undef,
+            last_matched_module_id => undef,
+            file_present           => 1,
+        });
+    my $real_needle = $needles->create(
+        {
+            dir_id                 => $real_needle_dir_id,
+            filename               => 'bootloader.json',
+            last_seen_module_id    => 10,
+            last_seen_time         => time2str('%Y-%m-%d %H:%M:%S', time - 100000),
+            last_matched_module_id => 9,
+            last_matched_time      => time2str('%Y-%m-%d %H:%M:%S', time - 50000),
+            file_present           => 1,
+            t_created              => time2str('%Y-%m-%d %H:%M:%S', time - 200000),
+            t_updated              => time2str('%Y-%m-%d %H:%M:%S', time - 200000),
+        });
+
+    goto_admin_needle_table();
+    $driver->find_element('.dataTables_filter input')->send_keys("bootloader");
+    wait_for_ajax;
+    my @needle_trs = $driver->find_elements('#needles tbody tr');
+    is(scalar(@needle_trs), 2, 'two added needles shown');
+    my @symlink_needle_tds = $driver->find_child_elements($needle_trs[1], 'td', 'css');
+    is((shift @symlink_needle_tds)->get_text(), 'symlink_needle_dir', 'symlink needle dir is right');
+    is((shift @symlink_needle_tds)->get_text(), 'bootloader.json',    'symlink needle file name is right');
+    is((shift @symlink_needle_tds)->get_text(), 'a day ago',          'symlink needle last use is right');
+    is((shift @symlink_needle_tds)->get_text(), 'about 14 hours ago', 'symlink needle last match is right');
```

Comment:
I guess we can find better explanations than "… is right". Can you be more specific?

Reply:
I changed `is right` to `is displayed correct`, does it make sense?  :)